### PR TITLE
Make the second homework

### PR DESCRIPTION
**Pull request by Sontsevolosa**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Correct solution) btw, it is preferable to use getter/setter instead of accessing instance variables directly. And in your initialize method try to use hsh.fetch to provide same functionality.

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -1,5 +1,27 @@
 # frozen_string_literal: true
 
-# exercise state and behaviour
 module StateAndBehaviour
+  # describe cars
+  class Car
+    attr_accessor(:year, :color, :model, :current_speed)
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: please read [parentheses section](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens) in style guide

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+      @color = characteristics[:color] || 'white'
+      @model = characteristics[:model] || 'Mustang'
+      @current_speed = 160
+    end
+
+    def push_break(speed)
+      @current_speed -= speed if @current_speed > speed
+    end
+
+    def self.default_car
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: and methods order section in style guide

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+      Car.new
+    end
+
+    def speed_up(speed)
+      @current_speed = speed
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: why do you use `@` if you have an attr_accessor?

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -1,5 +1,27 @@
 # frozen_string_literal: true
 
-# exercise state and behaviour
 module StateAndBehaviour
+  # describe cars
+  class Car
+    attr_accessor(:year, :color, :model, :current_speed)
+
+    def initialize(characteristics)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: `characteristics` are no common used name for attributes hash.
try: `car_options = {}`

File: lib/topics/state_and_behaviour.rb [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+      @year = characteristics[:year] || '1970'
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: at the class top:
`DEFAULT_YEAR = '1970'`
and fetch:
`@year = car_options.fetch(:year, DEFAULT_YEAR) # avoid magic values`

**Comment by Sontsevolosa** [comment]
I added only getter (without setter) in according to @sveredyuk 's decision.

@ghostinshadow @vbyno @sveredyuk please, review my code

File: lib/topics/state_and_behaviour.rb [hunk 6]
```diff
@@ -1,5 +1,29 @@
...
+
+    attr_reader :year, :color, :model, :current_speed
+
+    def self.default_car
+      Car.new
+    end
+
+    def initialize(car_options)
+      @year = car_options.fetch(:year, DEFAULT_YEAR)
+      @color = car_options.fetch(:color, 'white')
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb:16]: DEFAULT_COLOR

File: lib/topics/state_and_behaviour.rb [hunk 7]
```diff
@@ -1,5 +1,29 @@
...
+      @model = car_options.fetch(:model, 'Mustang')
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb:17]: DEFAULT_MODEL

File: lib/topics/state_and_behaviour.rb [hunk 8]
```diff
@@ -1,5 +1,29 @@
...
+      @current_speed = 160
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb:18]: DEFAULT_SPEED

**Pull request closed without merging**